refarch-integrations/refarch-s3-integration/refarch-s3-integration-client/src/main/java/de/muenchen/refarch/s3/integration/client/repository/DocumentStorageFileRepository.java

```diff
@@ -0,0 +1,195 @@
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestClientException;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class DocumentStorageFileRepository {
+
+    private final PresignedUrlRepository presignedUrlRepository;
+
+    private final S3FileTransferRepository s3FileTransferRepository;
+
+    private final ApiClientFactory apiClientFactory;
+
+    /**
+     * Gets the file specified in the parameter from the document storage.
+     *
+     * @param pathToFile defines the path to the file.
+     * @param expireInMinutes the expiration time of the presignedURL in minutes.
+     * @param documentStorageUrl to define to which document storage the request goes.
+     * @return the file.
+     * @throws DocumentStorageClientErrorException if the problem is with the client.
+     * @throws DocumentStorageServerErrorException if the problem is with the S3 storage or document
+     *             storage.
+     * @throws DocumentStorageException if the problem cannot be assigned to either the client or the S3
+     *             storage or the document storage.
+     */
+    public byte[] getFile(final String pathToFile, final int expireInMinutes, final String documentStorageUrl)
+            throws DocumentStorageException, DocumentStorageClientErrorException, DocumentStorageServerErrorException {
+        final Mono<String> presignedUrl = this.presignedUrlRepository.getPresignedUrlGetFile(pathToFile, expireInMinutes, documentStorageUrl);
+        return this.s3FileTransferRepository.getFile(presignedUrl.block());
+    }
+
+    /**
+     * Retrieves the file size of a file specified in the parameter from the document storage.
+     *
+     * @param pathToFile defines the path to the file.
+     * @param documentStorageUrl defines to which document storage the request goes.
+     * @return the file size.
+     * @throws DocumentStorageClientErrorException if the problem is with the client.
+     * @throws DocumentStorageServerErrorException if the problem is with the S3 storage or document
+     *             storage.
+     * @throws DocumentStorageException if the problem cannot be assigned to either the client or the S3
+     *             storage or the document storage.
+     */
+    public Mono<Long> getFileSize(final String pathToFile, final String documentStorageUrl)
+            throws DocumentStorageClientErrorException, DocumentStorageServerErrorException, DocumentStorageException {
+        try {
+            final FileApiApi fileApi = this.apiClientFactory.getFileApiForDocumentStorageUrl(documentStorageUrl);
+            return fileApi.getFileSize(pathToFile).mapNotNull(FileSizeDto::getFileSize);
+        } catch (final HttpClientErrorException exception) {
+            final String message = String.format("The request to get file size failed %s.", exception.getStatusCode());
```

Comment:
Maybe it would be a good idea to refactor the error messages to constants as in FileService? Applies to all Repository classes so please check carefully for error messages.

Reply:
In this case we could do it as it occurs multiple times, in general i like having the error messages directly where they are thrown.
So i would let it the way it is. But think that is also a personal practice thing.

Reply:
Since the messages are used multiple times I would thinks its better to use a constant field in this case. Otherwise PMD will throw a violation anyways later on when you have duplicate literals.

Reply:
I will then fix it when introducing pmd and not here, as the other integrations in DigiWF might have the same problem.